### DISCUSSION TITLE
Utility for hover & focus

### DISCUSSION THREAD

**Discussion by fs-c** [tailwindlabs/tailwindcss#4274] [Ideas]
I often find myself using `hover:` and `focus:` together (e. g. doing something like `hover:bg-gray-300 focus:bg-gray-300`). In fact, I don't think I ever used one of them without the other in my most recent project. Maybe a utility that combines them would be a good idea?

**Comment by JoshMoreno** [comment]
Here's what I came up with:
```js
plugins: [
    plugin(function({ addVariant, e }) {
        addVariant('hover|focus', ({ modifySelectors, separator }) => {
            modifySelectors(({ className }) => {
                const selector = `.${e(`hover|focus${separator}${className}`)}`
                return `${selector}:hover, ${selector}:focus`
            })
        })
    })
]
```
then add the variant to whatever you need
```js
variants: {
    extend: {
        backgroundColor: ['hover|focus'],
        textDecoration: ['hover|focus'],
        opacity: ['hover|focus']
    }
}
```

**Reply by simonswiss** [reply]
I've seen folks calling this state "hocus", and I don't hate it – it conveys the message of handling both states 😅
